core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -724,6 +756,23 @@ public void kill() throws InterruptedException {
                 int pid = getPid();
                 LOGGER.fine("Killing pid="+pid);
                 UnixReflection.destroy(pid);
+                // after sending SIGTERM, wait for the process to cease to exist
+                long deadline = System.nanoTime() + softkillWaitSeconds * 1000000000;
+                do {
+                    try {
+                        FileReader r = new FileReader(getFile("status"));
```

Comment:
Why not just `File#exists()`, what is the idea behind this reader logic?

Reply:
Cannot remember, probably a brain fart. exists should be enough indeed.